### PR TITLE
fix(server/pypika): Add double quotes around step name in WITH...AS statement for Snowflake

### PR DESCRIPTION
**Pull request by jGundermann**

### CODE REVIEW

File: server/src/weaverbird/backends/pypika_translator/translators/snowflake.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
-from typing import Any, TypeVar
+from typing import Any, TypeVar, Union
 
-from pypika.dialects import SnowflakeQueryBuilder
+from pypika.dialects import SnowflakeQueryBuilder, QueryBuilder
 from pypika.enums import Dialects
-from pypika.queries import Query
+from pypika.queries import Query, Selectable, Table, AliasedQuery
 from pypika.terms import Field, Term
+from pypika.utils import builder, format_alias_sql
```
Comment (lukapeschke) [inline server/src/weaverbird/backends/pypika_translator/translators/snowflake.py]:
```suggestion
from pypika.utils import builder, format_quotes, format_alias_sql
```

File: server/src/weaverbird/backends/pypika_translator/translators/snowflake.py [hunk 2]
```diff
@@ -19,6 +20,12 @@
 class QuotedSnowflakeQueryBuilder(SnowflakeQueryBuilder):
     QUOTE_CHAR = '"'
 
+    @builder
+    def with_(self, selectable: Selectable, name: str) -> "QueryBuilder":
+        t = AliasedQuery(f'"{name}"', selectable)
+        self._with.append(t)
```
Comment (lukapeschke) [inline server/src/weaverbird/backends/pypika_translator/translators/snowflake.py:26]:
```suggestion
        return super().with(selectable, format_quotes(name, self.QUOTE_CHAR))
```

File: server/src/weaverbird/backends/pypika_translator/translators/snowflake.py [hunk 3]
```diff
@@ -1,9 +1,10 @@
 from typing import Any, TypeVar
 
-from pypika.dialects import SnowflakeQueryBuilder
+from pypika.dialects import QueryBuilder, SnowflakeQueryBuilder
 from pypika.enums import Dialects
-from pypika.queries import Query
+from pypika.queries import AliasedQuery, Query, Selectable
 from pypika.terms import Field, Term
+from pypika.utils import builder
```
Comment (lukapeschke) [inline server/src/weaverbird/backends/pypika_translator/translators/snowflake.py:7]:
```suggestion
from pypika.utils import builder, format_quotes
```

**Review by lukapeschke — CHANGES_REQUESTED** [review]
Thanks for this! If this is needed for LTS, could you please target the `backend/release/v0.26` instead and backport it to master afterwards ? 

If not, you can keep master only.

Also could you please add a unit test ensuring we're doing this for snowflake only ?

Lastly, code LGTM but I have two minor suggestions: I'd rather leave as much as possible to PyPika, as the internals may change

**Comment by lukapeschke** [comment]
Oh and also add a changelog entry in `server/CHANGELOG.md` :pray:

**Comment by lukapeschke** [comment]
And also please change the commit message :smile: "double quotes" instead of "parentheses" + commitizen . Maybe something like `fix(server/pypika): Add double quotes around step name in WITH...AS statement for Snowflake`

File: server/src/weaverbird/backends/pypika_translator/translators/snowflake.py [hunk 4]
```diff
@@ -1,9 +1,10 @@
 from typing import Any, TypeVar
 
-from pypika.dialects import SnowflakeQueryBuilder
+from pypika.dialects import QueryBuilder, SnowflakeQueryBuilder
 from pypika.enums import Dialects
-from pypika.queries import Query
+from pypika.queries import AliasedQuery, Query, Selectable
```
Comment (Sanix-Darker) [inline server/src/weaverbird/backends/pypika_translator/translators/snowflake.py:5]:
```suggestion
from pypika.queries import Query, Selectable
```

Just following @lukapeschke review !

**Review by lukapeschke — APPROVED** [review]

File: server/src/weaverbird/backends/pypika_translator/translators/snowflake.py [hunk 5]
```diff
@@ -19,6 +20,11 @@
 class QuotedSnowflakeQueryBuilder(SnowflakeQueryBuilder):
     QUOTE_CHAR = '"'
 
+    @builder
+    def with_(self, selectable: Selectable, name: str) -> "QueryBuilder":
+        t = AliasedQuery(format_quotes(name, self.QUOTE_CHAR), selectable)
+        self._with.append(t)
```
Comment (lukapeschke) [inline server/src/weaverbird/backends/pypika_translator/translators/snowflake.py:26]:
```suggestion
        super().with_(selectable, format_quotes(name, self.QUOTE_CHAR))
```

File: server/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,9 @@
 # Changelog (weaverbird python package)
 
+## [0.26.2]
+
```
Comment (lukapeschke) [inline server/CHANGELOG.md:4]:
```suggestion
```

Will make backport easier

File: server/CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,7 @@
 # Changelog (weaverbird python package)
 
+- Pypika: Snowflake translator custom queryBuilder add's double quote's to with statement's
```
Comment (lukapeschke) [inline server/CHANGELOG.md:3]:
```suggestion
- Pypika: Snowflake translator custom queryBuilder adds double quotes to `WITH` statements
```

File: server/src/weaverbird/backends/pypika_translator/translators/snowflake.py (hunk 4 above)
Comment (Sanix-Darker) [inline server/src/weaverbird/backends/pypika_translator/translators/snowflake.py:5]:
```suggestion
from pypika.queries import Query, Selectable
```

**Review by Sanix-Darker — APPROVED** [review]

**Pull request merged**